src/librustc/traits/project.rs

```diff
@@ -1247,6 +1247,34 @@ fn confirm_callable_candidate<'cx, 'gcx, 'tcx>(
                                               fn_sig,
                                               flag);
 
+    if selcx.projection_mode().is_any() {
+        // Issue #33364: `ProjectionMode::Any` means trans, during
+        // which there is no (good) reason for confirmation to fail
+        // (because type check has already admitted the source code).
```

Comment:
I'm not sure in which situations this may backfire, but for example, vtable building in trans probes for obligations on methods to decide whether to include them. There are also secondary probes related to specialization, during selection.